freqtrade/data/converter/converter.py

```diff
@@ -44,6 +47,292 @@ def ohlcv_to_dataframe(ohlcv: list, timeframe: str, pair: str, *,
+    cols = DEFAULT_TRADES_COLUMNS
+    df = DataFrame(trades, columns=cols)
+    df['date'] = pd.to_datetime(
+        df['timestamp'], unit='ms', utc=True)
+
+    # Some exchanges return int values for Volume and even for OHLC.
+    # Convert them since TA-LIB indicators used in the strategy assume floats
+    # and fail with exception...
+    df = df.astype(dtype={'amount': 'float', 'cost': 'float',
+                          'price': 'float'})
+    return df
+
+
+def trades_to_volumeprofile_with_total_delta_bid_ask(trades: DataFrame, scale: float):
+    """
+    :param trades: dataframe
+    :param scale: scale aka bin size e.g. 0.5
+    :return: trades binned to levels according to scale aka orderflow
+    """
+    df = pd.DataFrame([], columns=DEFAULT_ORDERFLOW_COLUMNS)
+    # create bid, ask where side is sell or buy
+    df['bid_amount'] = np.where(
+        trades['side'].str.contains('buy'), 0, trades['amount'])
+    df['ask_amount'] = np.where(
+        trades['side'].str.contains('sell'), 0, trades['amount'])
+    df['bid'] = np.where(
+        trades['side'].str.contains('buy'), 0, 1)
+    df['ask'] = np.where(
+        trades['side'].str.contains('sell'), 0, 1)
+
+    # round the prices to the nearest multiple of the scale
+    df['price'] = ((trades['price'] / scale).round()
+                   * scale).astype('float64').values
+    if df.empty:
+        df['total'] = np.nan
+        df['delta'] = np.nan
+        return df
+
+    df['delta'] = df['ask_amount'] - df['bid_amount']
+    df['total_volume'] = df['ask_amount'] + df['bid_amount']
+    df['total_trades'] = df['ask'] + df['bid']
+
+    # group to bins aka apply scale
+    df = df.groupby('price').sum(numeric_only=True)
+    return df
+
+
+def trades_orderflow_to_imbalances(df: DataFrame, imbalance_ratio: int, imbalance_volume: int):
+    """
+    :param df: dataframes with bid and ask
+    :param imbalance_ratio: imbalance_ratio e.g. 300
+    :param imbalance_volume: imbalance volume e.g. 3)
+    :return: dataframe with bid and ask imbalance
+    """
+    bid = df.bid
+    ask = df.ask.shift(-1)
+    bid_imbalance = (bid / ask) > (imbalance_ratio / 100)
+    # overwrite bid_imbalance with False if volume is not big enough
+    bid_imbalance_filtered = np.where(
+        df.total_volume < imbalance_volume, False, bid_imbalance)
+    ask_imbalance = (ask / bid) > (imbalance_ratio / 100)
+    # overwrite ask_imbalance with False if volume is not big enough
+    ask_imbalance_filtered = np.where(
+        df.total_volume < imbalance_volume, False, ask_imbalance)
+    dataframe = DataFrame(
+        {"bid_imbalance": bid_imbalance_filtered,
+         "ask_imbalance": ask_imbalance_filtered},
+        index=df.index,
+    )
+
+    return dataframe
+
+
+def stacked_imbalance(df: DataFrame,
+                      label: str = "bid",
+                      stacked_imbalance_range: int = 3,
+                      should_reverse: bool = False):
```

Comment:
you should use defaults only if there's cases where you call it without the value - or if it's supposed to be called by a user - neither of which i think is the case here?